core/src/main/java/hudson/util/TextFile.java

```diff
@@ -57,8 +57,8 @@ public boolean exists() {
         return file.exists();
     }
 
-    public void delete() {
-        file.delete();
+    public void delete() throws IOException {
+        Files.deleteIfExists(Util.fileToPath(file));
     }
```

Comment:
Technically an API change, but there are no plugin consumers of this API, and all the consumers in core were already throwing `IOException` anyway.